db/channel_cache.go

```diff
@@ -371,9 +372,9 @@ func (c *channelCacheImpl) addChannelCache(channelName string) (*singleChannelCa
 	return singleChannelCache, true
 }
 
-func (c *channelCacheImpl) getActiveChannelCache(channelName string) (*singleChannelCacheImpl, bool) {
+func (c *channelCacheImpl) getActiveChannelCache(channel channels.ID) (*singleChannelCacheImpl, bool) {
 
-	cacheValue, found := c.channelCaches.Get(channelName)
+	cacheValue, found := c.channelCaches.Get(channel.String())
```

Comment:
This is actually performance sensitive and I decided to make `RangeSafeCollection` not generic in order to support this. I think that I would just have to change the iteration before to make this work on `interface{}` and this might actually be a use for go generics.